Content.Server/VendingMachineRestockPackage/VendingMachineRestockPackageSystem.cs

```diff
@@ -0,0 +1,159 @@
+using Content.Server.Wires;
+using Content.Shared.Interaction;
+using Content.Shared.Popups;
+using Content.Shared.VendingMachines;
+
+namespace Content.Server.VendingMachineRestockPackage
+{
+    public sealed class VendingMachineRestockPackageSystem : EntitySystem
+    {
+        [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+        [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+        [Dependency] private readonly AudioSystem _audioSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<VendingMachineRestockPackageComponent, AfterInteractEvent>(OnAfterInteract);
+
+            SubscribeLocalEvent<RestockSuccessfulEvent>(OnRestockSuccessful);
+            SubscribeLocalEvent<RestockCancelledEvent>(OnRestockCancelled);
+        }
+
+        public bool TryAccessMachine(EntityUid uid,
+            VendingMachineRestockPackageComponent component,
+            VendingMachineComponent machineComponent,
+            EntityUid user,
+            EntityUid target)
+        {
+            if (!TryComp<WiresComponent>(target, out var wires) || !wires.IsPanelOpen) {
+                _popupSystem.PopupCursor(Loc.GetString("vending-machine-restock-package-needs-panel-open",
+                        ("this", uid),
+                        ("user", user),
+                        ("target", target)
+                        ),
+                    Filter.Entities(user));
+                return false;
+            }
+
+            return true;
+        }
+
+        public bool TryMatchPackageToMachine(EntityUid uid,
+            VendingMachineRestockPackageComponent component,
+            VendingMachineComponent machineComponent,
+            EntityUid user,
+            EntityUid target)
+        {
+            if (!component.CanRestock.Contains(machineComponent.PackPrototypeId)) {
+                _popupSystem.PopupCursor(Loc.GetString("vending-machine-restock-package-invalid-inventory",
+                        ("this", uid),
+                        ("user", user),
+                        ("target", target)
+                        ),
+                    Filter.Entities(user));
+                return false;
+            }
+
+            return true;
+        }
+
+        private void OnAfterInteract(EntityUid uid, VendingMachineRestockPackageComponent component, AfterInteractEvent args)
+        {
+            if (component.CancelToken != null || args.Target == null || !args.CanReach)
+                return;
+
+            if (!TryComp<VendingMachineComponent>(args.Target, out var machineComponent))
+                return;
+
+            if (!TryMatchPackageToMachine(uid, component, machineComponent, args.User, args.Target.GetValueOrDefault()))
+                return;
+
+            if (!TryAccessMachine(uid, component, machineComponent, args.User, args.Target.GetValueOrDefault()))
+                return;
```

Comment:
```suggestion
            if (!TryMatchPackageToMachine(uid, component, machineComponent, args.User, args.Target.Value))
                return;

            if (!TryAccessMachine(uid, component, machineComponent, args.User, args.Target.Value))
                return;
```

target was already null checked, so this works fine